### PR TITLE
enh(CPlusPlus): memory usage on Dijksta algorithm

### PR DESCRIPTION
**Pull request by Denisthatorf**
https://github.com/MakeContributions/DSA/issues/1058 - try to fix this problem
I optimize the Dijkstra algorithm in C++ 

The Dijkstra algorithm uses vector<vector<pair<int,int>>> graph(V+1,vector<pair<int,int>>()) as a matrix. This is very, very bad for memory, because the vector in the vector in memory is not stored linearly, and can be scattered in memory very much. As an alternative, I propose to use a regular array (data = new T [rows * columns]) allocated on the heap and get its elements like this : *(_data + row * _rows + column).

### CODE REVIEW

File: algorithms/CPlusPlus/Graphs/dijkstra.cpp [hunk 1]
```diff
@@ -1,99 +1,177 @@
...
-        //otherwise we update our current cost(distance)
-        dist[cVertex]=cCost;
-
-        if(cVertex==end){
-            ret=cCost;
-            break;
+#include <cstddef>
+#include <limits>
+
+//I highly recomend to create matrix class
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Graphs/dijkstra.cpp:4]:
```suggestion
//I highly recommend to create matrix class
```

File: algorithms/CPlusPlus/Graphs/dijkstra.cpp [hunk 2]
```diff
@@ -1,99 +1,177 @@
...
+#include <iostream>
+
+
+//function declaration below
+//Generate and std::cout << matrix
+void getExampleMatrix(int*& matrix_out, size_t& size_out);
+
+int main()
+{
+    //I highly recomend to create matrix class
```
Comment (Arsenic-ATG) [inline algorithms/CPlusPlus/Graphs/dijkstra.cpp:85]:
```suggestion
    //I highly recommend to create matrix class
```

**Review by Arsenic-ATG — CHANGES_REQUESTED** [review]
typos causing codespell check to fail 

```
./algorithms/CPlusPlus/Graphs/dijkstra.cpp:4: recomend ==> recommend
./algorithms/CPlusPlus/Graphs/dijkstra.cpp:85: recomend ==> recommend
```

**Comment by Denisthatorf** [comment]
Can you check one more time?

**Pull request merged**